src/services/oracle.ts

```diff
@@ -37,36 +36,18 @@ export const OracleAbi = [
  */
 export class OracleService<T extends ChainId> extends ContractService<T> {
   static abi = OracleAbi;
+  static contractId = ContractAddressId.oracle;
 
-  constructor(chainId: T, ctx: Context) {
-    super(ctx.addresses.oracle ?? OracleService.addressByChain(chainId), chainId, ctx);
+  get contract() {
+    return this._getContract(OracleService.abi, OracleService.contractId, this.ctx);
   }
-
-  /**
-   * Get most up-to-date address of the Oracle contract for a particular chain
-   * id.
-   * @param chainId
-   * @returns address
-   */
-  static addressByChain(chainId: ChainId): string {
-    switch (chainId) {
-      case 1:
-      case 1337:
-        return "0x83d95e0D5f402511dB06817Aff3f9eA88224B030";
-      case 250:
-        return "0x57AA88A0810dfe3f9b71a9b179Dd8bF5F956C46A";
-      case 42161:
-        return "0x043518AB266485dC085a1DB095B8d9C2Fc78E9b9";
-    }
-  }
-
   /**
    * Fetch all the active Oracle calculations.
    * @param overrides
    * @returns list of calculations contract addresses
    */
   async getCalculations(overrides: CallOverrides = {}): Promise<Address[]> {
-    return await this.contract.read.calculations(overrides);
+    return await this.contract.then(contract => contract.read.calculations(overrides));
```

Comment:
We are missing the refactor of defining `const contract = await this.contract;` on this file